src/main/java/store/domain/order/Order.java

```diff
@@ -0,0 +1,34 @@
+package store.domain.order;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Order {
+    private final List<OrderItem> orders;
+
+    public Order(final List<OrderItem> orders) {
+        this.orders = orders;
+    }
+
+    public List<OrderItem> findOrderItemHavingPromotion() {
+        List<OrderItem> promotionAvailableOrderItems = new ArrayList<>();
+        for (OrderItem orderItem : orders) {
+            if (orderItem.isOrderProductHavePromotion()) {
+                promotionAvailableOrderItems.add(orderItem);
+            }
+        }
+        return promotionAvailableOrderItems;
```

Comment:
아래와 같이 stream으로 변환하시면 가독성 측면에서 더 좋을 것 같아요!

```suggestion
        return orders.stream()
                .filter(OrderItem::isOrderProductHavePromotion)
                .collect(Collectors.toList());
```